doc.go

```diff
@@ -147,9 +147,15 @@
 //      CheckOrigin: func(r *http.Request) bool { return true },
 //  }
 //
-// The deprecated Upgrade function does not enforce an origin policy. It's the
-// application's responsibility to check the Origin header before calling
-// Upgrade.
+// We recommend the upgrader.Upgrade method to perform an upgrade
```

Comment:
But, the key idea here is that it's the only non-deprecated mechanism for upgrading a connection. It isn't the only mechanism for upgrading a connection. Thus, due to Upgrade's deprecation we recommend Upgrader.Upgrade.